### PR TITLE
add API for method create MCif

### PR DESCRIPTION
**Pull request by kumahauzi**
Screenshot (154)

### CODE REVIEW

**Comment by kumahauzi** [comment]
request:
{
    "customerName": "entud",
    "dateOfBirth": "2007-12-14",
    "address": "dago",
    "phoneNumber": "09857575758787",
    "email": "entud@gmail.com",
    "idNumber": "7439779799777",
    "idNumberType": "1"
}
response : 
{
    "cifId": "4028e49293c09c600193c09cdf050000",
    "customerName": "entud",
    "dateOfBirth": null,
    "address": "dago",
    "phoneNumber": "09857575758787",
    "email": "entud@gmail.com",
    "idNumber": "7439779799777",
    "idNumberType": "7439779799777",
    "createdAt": null,
    "upDateAt": null,
    "createdBy": null,
    "upDateBy": null,
    "authorizationAt": null,
    "authorizationBy": null,
    "statusId": "1",
    "statusName": null,
    "deleted": false
}

File: src/main/java/com/b2camp/simple_core_banking/controller/TSavingAccountController.java [hunk 1]
```diff
@@ -1,16 +1,35 @@
...
 public class TSavingAccountController {
     private TSavingAccountService tSavingAccountService;
 
+    @Autowired
     TSavingAccountController(TSavingAccountService tSavingAccountService) {
         this.tSavingAccountService = tSavingAccountService;
     }
+
+    @GetMapping
+    List<TSavingAccountResponse>readreadTsavingAccount(@Param("accountNumber")String accountNumber){
```
Comment (xuibdg) [inline src/main/java/com/b2camp/simple_core_banking/controller/TSavingAccountController.java]: rename method, Do you think the name double read is correct?

File: src/main/java/com/b2camp/simple_core_banking/repository/TSavingAccountRepository.java [hunk 2]
```diff
@@ -1,7 +1,22 @@
...
+import java.util.List;
+import java.util.Optional;
+
+@Repository
 public interface TSavingAccountRepository extends JpaRepository<TSavingAccount, String> {
+
+    @Query("SELECT tsa FROM TSavingAccount tsa WHERE (:accountNumber IS NULL OR :accountNumber = '' OR LOWER(tsa.accountNumber) LIKE LOWER(CONCAT('%', :accountNumber, '%'))) AND tsa.isDeleted = false")
+    List<TSavingAccount> findAllByAccountNumberAndIsDeletedIsFalse(@Param("accountNumber") String accountNumber);
+
+    Optional<TSavingAccount>findBySavingId(String sevingId);
```
Comment (xuibdg) [inline src/main/java/com/b2camp/simple_core_banking/repository/TSavingAccountRepository.java]: what is param sevingId? rename param

File: src/main/java/com/b2camp/simple_core_banking/service/TSavingAccountService.java [hunk 3]
```diff
@@ -1,4 +1,14 @@
...
 
+import com.b2camp.simple_core_banking.dto.TSavingAccountResponse;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
 public interface TSavingAccountService {
+    List<TSavingAccountResponse>readTsavingAccount(String accountNumber);
```
Comment (xuibdg) [inline src/main/java/com/b2camp/simple_core_banking/service/TSavingAccountService.java]: rename to camelcase

File: src/main/java/com/b2camp/simple_core_banking/service/impl/MCifServiceImpl.java [hunk 4]
```diff
@@ -1,7 +1,74 @@
...
+    @Transactional
+    public MCifResponse createCif(MCifRequest request) {
+        MCif mCif = new MCif();
+        buildEntity(mCif, request);
+        mCifRepository.save(mCif);
+
+        return buildResponseCreate(mCif);
+    }
+
+    private MCifResponse buildResponseCreate(MCif mCif) {
```
Comment (xuibdg) [inline src/main/java/com/b2camp/simple_core_banking/service/impl/MCifServiceImpl.java]: rename to buildResponseForCreate

File: src/main/java/com/b2camp/simple_core_banking/service/impl/TSavingAccountServiceImpl.java [hunk 5]
```diff
@@ -1,6 +1,72 @@
...
+    @Override
+    public Optional<TSavingAccountResponse> findBySavingAccountId(String savingAccountId) {
+        TSavingAccount tSavingAccount = tSavingAccountRepository.findById(savingAccountId)
+                .orElseThrow(()-> new RuntimeException("Data saving id tidak ada: " + savingAccountId));
+        return Optional.of(buildRespons(tSavingAccount));
+    }
+
+
+
+    private TSavingAccountResponse buildRespons(TSavingAccount tSavingAccount) {
```
Comment (xuibdg) [inline src/main/java/com/b2camp/simple_core_banking/service/impl/TSavingAccountServiceImpl.java]: rename to buildResponse

**Review by Yaksyin — APPROVED** [review]

**Review by xuibdg — APPROVED** [review]

**Pull request merged**